devicetypes/smartthings/zwave-basic-heat-alarm.src/zwave-basic-heat-alarm.groovy

```diff
@@ -0,0 +1,165 @@
+				attributeState("detected", label: "HEAT", icon: "st.alarm.smoke.smoke", backgroundColor: "#e86d13")
+				attributeState("tested", label: "TEST", icon: "st.alarm.smoke.test", backgroundColor: "#e86d13")
+			}
+		}
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+
+		main "heat"
+		details(["heat", "battery"])
+	}
+}
+
+def installed() {
+	// Device checks in every hour, this interval allows us to miss one check-in notification before marking offline
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	def cmds = []
+	createHeatEvents("heatClear", cmds)
+	cmds.each { cmd -> sendEvent(cmd) }
+}
+
+def updated() {
+	// Device checks in every hour, this interval allows us to miss one check-in notification before marking offline
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def getCommandClassVersions() {
+	[
+			0x80: 1, // Battery
+			0x84: 1, // Wake Up
+			0x71: 3, // Alarm
+			0x72: 1, // Manufacturer Specific
+	]
+}
+
+def parse(String description) {
+	def results = []
+	if (description.startsWith("Err")) {
+		results << createEvent(descriptionText: description, displayed: true)
+	} else {
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			results += zwaveEvent(cmd)
+		}
+	}
+	log.debug "'$description' parsed to ${results.inspect()}"
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	def results = []
+	results << createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)
+	if (!state.lastbatt || (now() - state.lastbatt) >= 56 * 60 * 60 * 1000) {
+		results << response([
+				zwave.batteryV1.batteryGet().format(),
+				"delay 2000",
+				zwave.wakeUpV1.wakeUpNoMoreInformation().format()
+		])
+	} else {
+		results << response(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	}
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%", isStateChange: true]
+	state.lastbatt = now()
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "$device.displayName battery is low!"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	return createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	def event = [displayed: false]
+	event.linkText = device.label ?: device.name
+	event.descriptionText = "$event.linkText: $cmd"
+	return createEvent(event)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def result = null
+	if (cmd.notificationType == 0x04) {  // Heat Alarm
+		switch (cmd.event) {
+			case 0x00:
+			case 0xFE:
+				result = createHeatEvents("heatClear")
```

Comment:
I have a similar comment here as I did for the smoke detector changes.

Namely that 0x00 should clear both "heat" and "tested" and that 0xFE shouldn't do anything.

Reply:
Actually, nevermind. I realize now that all of this is only for the single "heat" variable that can be either "detected", "clear", or "test" and that "heatClear" and "testClear" are functionally identical.

Reply:
OK. No changes required then.